vignettes/pdp.Rmd

```diff
@@ -0,0 +1,103 @@
+---
+title: "Partial Dependence Explanations"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Partial Dependence Explanations}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+This vignette demonstrates how to use the Partial Dependence explanations in `survex`, as well as Accumulated Local Effects explanations. It especially demonstrates the usage of new kinds of plots that are available in the 1.1 version of the package. To create these explanations we follow the standard way of working with survex i.e. we create a model, and an explainer.
+
+```{r setup}
+library(survex)
+library(survival)
+library(ranger)
+
+set.seed(123)
+
+vet <- survival::veteran
+
+rsf <- ranger(Surv(time, status) ~ ., data = vet)
+exp <- explain(rsf, data = vet[, -c(3,4)], y = Surv(vet$time, vet$status))
+
+```
+
+We use the explainer and the `model_profile()` function to calculate Partial Dependence explanations. We can specify the variables for which we want to calculate the explanations. In this example we calculate the explanations for the variables `karno` and `celltype`. **Note:** The background for generating PD values is the `data` field of the explainer! If you want to calculate explanations with a background that is not the training data, you need to manually specify the `data` argument, when creating the explainer.
+
+We can calculate Accumulated Local Effects in the same way, by setting the `type` argument to `"accumulated"`.
+
+```{r}
+pdp <- model_profile(exp, variables = c("karno", "celltype"), n = 20)
+ale <- model_profile(exp, variables = c("karno", "celltype"), n = 20, type = "accumulated")
```

Comment:
I noticed a bug(?) when using 

ale <- model_profile(exp, type = „accumulated")
Error in rowSums(abs(A[dists[, "from.level"], ] - A[dists[, "to.level"],  : 
  'x' must be an array of at least two dimensions

on an explainer not generated based on the veteran dataset. Only occurs of no "variables" argument is passed to the function, error seems to originate in the order_levels function.

Reply:
Hmmm, I wasn't able to reproduce this error 😕

However, I've tried on 6 other datasets and subsets of them and spot another bug (resulting from the presence of character variables that are not factors or labelled as additional categorical variables - fixed). Errors can also be associated with the presence of only one unique value of a given variable in the data, but this causes other error messages.  

Could you check where exactly the error was coming from, or send the code snippet with the data, model and explainer creation?

Reply:
I attempted to recreate it now on the same data and it does not occur anymore. Either it was some conflict in my R environment or it was fixed by proxy it seems. Thank you for checking!